### PR TITLE
Fix inconsistencies & clean code

### PR DESCRIPTION
**Pull request by JonDum**
Inconsistent semicolons, duplicate variables, unused variables, dot notation, etc.

JSHint complained a shit ton about this:

25 warnings
35	Missing semicolon.
39	Missing semicolon.
42	Missing semicolon.
60	Use '!==' to compare with 'null'.
87	['display'] is better written in dot notation.
88	['display'] is better written in dot notation.
89	['display'] is better written in dot notation.
94	['display'] is better written in dot notation.
106	['transform'] is better written in dot notation.
111	Missing semicolon.
137	Missing semicolon.
142	Missing semicolon.
144	Missing semicolon.
249	['opacity'] is better written in dot notation.
250	['position'] is better written in dot notation.
251	['left'] is better written in dot notation.
252	['top'] is better written in dot notation.
256	['transform'] is better written in dot notation.
276	Missing semicolon.
296	Do not use 'new' for side effects.
298	Missing semicolon.
309	Missing semicolon.
313	Missing semicolon.
323	Missing semicolon.
353	'i' is already defined.
Two unused variables
75	index
303	el

### CODE REVIEW

File: ios-drag-drop.js [hunk 1]
```diff
@@ -292,7 +293,7 @@
         evt.preventDefault();
         new DragDrop(evt,el);
```
Comment (JonDum) [inline ios-drag-drop.js]: JSHint also doesn't like the way this is called though, "bare" `new` statement.

**Comment by xStrom** [comment]
Looks nice. I've had this on my todo list for the longest time and I guess I don't have to do it myself now.

**Comment by JonDum** [comment]
Open source ftw

**Comment by timruffles** [comment]
What's weird is you didn't use the `.jshintrc` in the project :)

File: ios-drag-drop.js [hunk 2]
```diff
@@ -316,7 +318,8 @@
   }
 
   function once(el, event, handler, context) {
-    if(context) handler = handler.bind(context)
+    if(context)
+      handler = handler.bind(context);
```
Comment (timruffles) [inline ios-drag-drop.js]: Can you revert this please, or give it brackets? This style of if statement is very error prone when people implicitly assume JS has significant whitespace.

File: ios-drag-drop.js [hunk 3]
```diff
@@ -82,14 +82,14 @@
       this.synthesizeEnterLeave(event);
     },
     hideDragImage: function() {
-      if (this.dragImage && this.dragImage.style["display"] != "none") {
-        this.dragImageDisplay = this.dragImage.style["display"];
-        this.dragImage.style["display"] = "none";
+      if (this.dragImage && this.dragImage.style.display != "none") {
```
Comment (timruffles) [inline ios-drag-drop.js]: missing `===` :)

File: ios-drag-drop.js (hunk 1 above)
Comment (timruffles) [inline ios-drag-drop.js]: sure, pop in a local var if you like. V8 will realise it's cruft and optimise away.

File: ios-drag-drop.js (hunk 2 above)
Comment (JonDum) [inline ios-drag-drop.js]: Which would you prefer? Single line w/ semicolon or brackets? I thought it was a mistake at first with it being an if statement on a single line with no semicolon.

**Comment by timruffles** [comment]
Thanks again for this, all in all LGTM! If you wouldn't mind making the `if` changes, and perhaps updating `.jshintrc` to enforce semis that'd be great.

File: ios-drag-drop.js (hunk 2 above)
Comment (timruffles) [inline ios-drag-drop.js]: either, just not the error prone style :)

**Comment by JonDum** [comment]
Fixed the if statements.

About the jshintrc, it's fine to have or not have semi's. This shim had more semi's than naked statements  so I just went with the least changes by adding the missing semi's.

**Comment by timruffles** [comment]
Sure! I wrote the original without but accepted a bunch of patches with.

Thanks again for the cleanup! If you've had a chance to test I'll merge.
Otherwise will merge once I've had a test.

**Comment by timruffles** [comment]
Landed - thanks again!

**Pull request merged**